src/Raven.Server/Documents/Indexes/Static/JavaScriptMapOperation.cs

```diff
@@ -155,8 +155,21 @@ public void Analyze(Engine engine)
 
                         if (IsBoostExpression(ce))
                             HasBoostedFields = true;
-                        else
+                        else if (IsCreateDynamicExpression(ce))
                             HasDynamicReturns = true;
+                        else if (ce.Arguments.Count == 1 && ce.Arguments.AsNodes()[0] is ArrowFunctionExpression afe && afe.Body is ObjectExpression oea)
```

Comment:
let's wrap this condition into a method with meaningful name

Reply:
Done